src/main.cpp

```diff
@@ -5728,7 +5730,7 @@ StructCPID GetLifetimeCPID(const std::string& cpid, const std::string& sCalledFr
         CBlockIndex* pblockindex = mapItem->second;
         if(pblockindex == NULL ||
            pblockindex->IsInMainChain() == false ||
-           pblockindex->GetCPID() != cpid)
+           pblockindex->cpid != cpid128)
```

Comment:
This will provide different (undefined?) results for non-researcher blocks. Please check the block flags before working with cpid field, to ensure its validity.
Note: cpid128 is fine, because of check at [L5707](https://github.com/gridcoin-community/Gridcoin-Research/pull/1206/files#diff-7ec3c68a81efff79b6ca22ac1f1eabbaR5709), but accessing cpid of the blockindex is invalid.

Reply:
You are absolutely right, that is the correct way of avoiding issues when the CPID actually is `000000...`. I'll add a check for it, thanks.